applications/StructuralMechanicsApplication/custom_constitutive/linear_j2_plasticity_3d.cpp

```diff
@@ -0,0 +1,464 @@
+        rValue = mStrainEnergy;
+    }
+    return(rValue);
+}
+
+void LinearJ2Plasticity3D::FinalizeMaterialResponsePK1(Parameters& rValues)
+{
+}
+
+void LinearJ2Plasticity3D::FinalizeMaterialResponsePK2(Parameters& rValues)
+{
+}
+
+void LinearJ2Plasticity3D::FinalizeMaterialResponseKirchhoff(Parameters& rValues)
+{
+}
+
+void LinearJ2Plasticity3D::FinalizeMaterialResponseCauchy(Parameters& rValues)
+{
+}
+
+double LinearJ2Plasticity3D::GetSaturationHardening(const Properties& rMaterialProperties)
+{
+    double yield_stress = rMaterialProperties[YIELD_STRESS];
+    double theta = rMaterialProperties[REFERENCE_HARDENING_MODULUS];
+    double hardening_modulus = rMaterialProperties[ISOTROPIC_HARDENING_MODULUS];
+    double delta_k = rMaterialProperties[INFINITY_HARDENING_MODULUS];
+    double hardening_exponent = rMaterialProperties[HARDENING_EXPONENT];
+    double k_new = yield_stress + (theta * hardening_modulus * mAccumulatedPlasticStrain) +
+                delta_k * (1. - std::exp(-hardening_exponent * mAccumulatedPlasticStrain));
+    return k_new;
+}
+
+double LinearJ2Plasticity3D::GetPlasticPotential(const Properties& rMaterialProperties)
+{
+    double theta = rMaterialProperties[REFERENCE_HARDENING_MODULUS];
+    double hardening_modulus = rMaterialProperties[ISOTROPIC_HARDENING_MODULUS];
+    double delta_k = rMaterialProperties[INFINITY_HARDENING_MODULUS];
+    double hardening_exponent = rMaterialProperties[HARDENING_EXPONENT];
+
+    double Wp_new = 0.5*(theta * hardening_modulus * std::pow(mAccumulatedPlasticStrain, 2.0)) +
+                    delta_k * (mAccumulatedPlasticStrain -
+                            (1/hardening_exponent) * (1- std::exp(-hardening_exponent * mAccumulatedPlasticStrain)));
+    return Wp_new;
+}
+
+double LinearJ2Plasticity3D::GetDeltaGamma(double norm_s_trial,
+                                           const Properties& rMaterialProperties)
+{
+    double E = rMaterialProperties[YOUNG_MODULUS];
+    double poisson_ratio = rMaterialProperties[POISSON_RATIO];
+    double yield_stress = rMaterialProperties[YIELD_STRESS];
+    double theta = rMaterialProperties[REFERENCE_HARDENING_MODULUS];
+    double hardening_modulus = rMaterialProperties[ISOTROPIC_HARDENING_MODULUS];
+    double delta_k = rMaterialProperties[INFINITY_HARDENING_MODULUS];
+    double hardening_exponent = rMaterialProperties[HARDENING_EXPONENT];
+    double tolerance = 1e-6 * yield_stress;
+    double mu = E / (2. * (1. + poisson_ratio));
+    double dgamma = 0.0;
+    double norm_yieldfunction = 1.0;
+
+    while (norm_yieldfunction > tolerance)
+    {
+        double k_new = GetSaturationHardening(rMaterialProperties);
+        double kp_new =
+            theta * hardening_modulus +
+            delta_k * (hardening_exponent *
+                       std::exp(-hardening_exponent * mAccumulatedPlasticStrain));
+        double yieldfunction = -0.8164965809277260 * k_new + norm_s_trial - 2. * mu * dgamma;
```

Comment:
Missing const's

Define -0.8164965809277260 as a double variable, to avoid problems in the value if the code is changed later